server/src/main/java/org/opensearch/search/backpressure/stats/SearchBackpressureStats.java

```diff
@@ -22,21 +23,34 @@
  * Stats related to search backpressure.
  */
 public class SearchBackpressureStats implements ToXContentFragment, Writeable {
-    private final SearchShardTaskStats searchShardTaskStats;
+    private final SearchBackpressureTaskStats searchTaskStats;
+    private final SearchBackpressureTaskStats searchShardTaskStats;
```

Comment:
For now, we don't have different stats structures for different task types so we will continue using the same class to avoid repetition. However, in future we can separate stats classes for the types of tasks.

Reply:
I still feel that the existing structure was already future-proof to handle this scenario. Tying them together now makes it harder to decouple later (eg. BWC needs to be handled correctly, it's non-trivial for future developers to decouple stats at this place, etc.).

Reply:
Got the point on BWC issues this might cause in future. Will separate out the stats classes for `SearchTaskStats` and `SearchShardTaskStats`.